### PR TITLE
Move ephemeral cluster from jumphost to internal server

### PR DESCRIPTION
**Pull request by Sunnatillo**
This PR moves running ephemeral cluster from jumphost to internal server in bml. 
1. Jumphost is critical for our lab. Dependencies of project always change and require newer package versions. It creates a risk of breaking OS at any point. OS that we test in CI already newer version than in jumphost. We have already seen package conflicts several times. 
2. After changes Ephemeral cluster (minikube) runs on internal server. We still use jumphost as a proxy to reach internal server.

### CODE REVIEW

**Review by tuminoid — COMMENTED** [review]
lgtm, but someone with more bml knowledge should actually lgtm this

**Review by furkatgofurov7 — COMMENTED** [review]
Changes look fine to me, however I am not fully aware of the 2 things with this PR, so if you could add/expand it in PR description, it would be very helpful:
1. Why do we need this change and what is the benefit? 
2. What would be different after this lands? Do we still keep using jumphost as a proxy between the user and new internal server?

**Comment by Sunnatillo** [comment]
> Changes look fine to me, however I am not fully aware of the 2 things with this PR, so if you could add/expand it in PR description, it would be very helpful:
> 
>     1. Why do we need this change and what is the benefit?
> 
>     2. What would be different after this lands? Do we still keep using jumphost as a proxy between the user and new internal server?

Updated

File: jenkins/scripts/bml_integration_test.sh
Comment (furkatgofurov7) [inline jenkins/scripts/bml_integration_test.sh:46]: I think we need to make sure that file description on the script usage in the beginning of this file is also up-to-date with the current changes (GitHub does not allow me to comment directly on that lines)

**Review by furkatgofurov7 — COMMENTED** [review]
Have not checked the log fetching myself for the BML run, but is it working properly? 
I don't see it being changed with current PR but we are specifying `TEST_EXECUTER_IP` in https://github.com/metal3-io/project-infra/blob/1285ef122e18ac40309e49505f60e71cf6536707/jenkins/scripts/fetch_logs.sh#L37

File: jenkins/scripts/bml_integration_test.sh
Comment (Sunnatillo) [reply jenkins/scripts/bml_integration_test.sh:46]: Good catch. I will add the changes for fetching logs script as well

File: jenkins/scripts/fetch_logs.sh
Comment (tuminoid) [inline jenkins/scripts/fetch_logs.sh:54]: This is excessively repetitive and error-prone way to call ssh/scp.

I'd suggest defining the common options something like this:

```bash
declare -a SSH_OPTIONS=(
    -o StrictHostKeyChecking=no
    -o UserKnownHostsFile=/dev/null
    -o ServerAliveInterval=15
    -i "${METAL3_CI_USER_KEY}"
    -o ProxyCommand="ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -i ${METAL3_CI_USER_KEY} -W %h:%p ${METAL3_CI_USER}@${TEST_EXECUTER_IP}"
)
```
and then all the calls would looks like
```bash
ssh \
  "${SSH_OPTIONS[@]}" \
  <user>@<port> \
  <whatever command>
```

File: jenkins/scripts/bml_integration_test.sh
Comment (tuminoid) [inline jenkins/scripts/bml_integration_test.sh:74]: Same comment applies here.

**Review by tuminoid — CHANGES_REQUESTED** [review]

File: jenkins/scripts/fetch_logs.sh
Comment (Sunnatillo) [reply jenkins/scripts/fetch_logs.sh]: Thank you for pointing it out.
Changed it according your suggestion.

File: jenkins/scripts/bml_integration_test.sh
Comment (Sunnatillo) [reply jenkins/scripts/bml_integration_test.sh]: done

**Review by tuminoid — APPROVED** [review]
/lgtm

File: jenkins/scripts/bml_integration_test.sh
Comment (furkatgofurov7) [reply jenkins/scripts/bml_integration_test.sh]: Looks like logs collected properly now

**Review by furkatgofurov7 — APPROVED** [review]
/approve
/hold leaving it to @kashifest to take a second look and unhold if looks good.

**Review by kashifest — COMMENTED** [review]
/hold cancel

**Pull request merged**
